sdk/core/azure-core/src/main/java/com/azure/core/util/Configuration.java

```diff
@@ -496,6 +496,38 @@ public boolean contains(ConfigurationProperty<?> property) {
      * @throws RuntimeException when property value conversion (and validation) throws.
      */
     public <T> T get(ConfigurationProperty<T> property) {
+        return get(property, false);
+    }
+
+    /**
+     * Gets property value from all available sources in the following order:
+     *
+     * <ul>
+     *     <li>Explicit configuration from given {@link ConfigurationSource} by property name</li>
+     *     <li>Explicit configuration by property aliases in the order they were provided in {@link ConfigurationProperty}</li>
+     *     <li>Explicit configuration by property name in the shared section (if {@link ConfigurationProperty} is shared)</li>
+     *     <li>Explicit configuration by property aliases in the shared section (if {@link ConfigurationProperty} is shared)</li>
+     *     <li>System property (if set)</li>
+     *     <li>Environment variable (if set)</li>
+     * </ul>
+     *
+     * <p>
+     * Property value is converted to specified type. If property value is missing and not required, default value is
+     * returned.
+     * <p>
+     * If the property value conversion fails, {@code returnDefaultOnConversionFailure} determines whether the default
+     * value is returned or the conversion exception is thrown.
+     *
+     * @param property instance.
+     * @param returnDefaultOnConversionFailure If true, returns the default value when the property value conversion
+     * fails. If false, throws an exception when the property value conversion fails.
+     * @param <T> Type that the configuration is converted to if found.
+     * @return The value of the property if it exists, otherwise the default value of the property.
+     * @throws NullPointerException when property instance is null.
+     * @throws IllegalArgumentException when required property is missing.
+     * @throws RuntimeException when property value conversion (and validation) throws.
+     */
+    public <T> T get(ConfigurationProperty<T> property, boolean returnDefaultOnConversionFailure) {
```

Comment:
There are just 2 config properties in the repo to which it could apply (AMQP and HTTP proxy options port).

I would not mind if we changed the existing `get(property)` to return default value if conversion fails (and throw if there is no default value) instead of adding a new method.
It could be very mildly breaking, but I don't expect any significant usage of ConfigProperties.

Otherwise, I'd add the flag to `ConfigPropertyBuilder` (`defaultValue(value, returnDefaultOnConversionFailure)`) instead of `Configuration.get`.